File: EcoMonitor.Api/Models/Measurement.cs
using System.ComponentModel.DataAnnotations;

namespace EcoMonitor.Api.Models;

public class Measurement
{
	public int Id{get;set;}
	public DateTime CreationDate{ get; set; } = DateTime.UtcNow;
	public int LocationId{get;set;}

	public int CreatorId{get;set;}
	[MaxLength(400)]
	public string Comment{get;set;} = string.Empty;
	public string SensorName{get;set;} = null!;

	public double? O2 { get; set; }
    	public double? CO { get; set; }
    	public double? SO2 { get; set; }
    	public double? NO { get; set; }
    	public double? CH { get; set; }
    	public double? CO2 { get; set; }
    	public double? NO2 { get; set; }
    	public double? H2CO { get; set; }
    	public double? PM25 { get; set; }
    	public double? PM10 { get; set; }
    	public double? TVOC { get; set; }

    	public double? WindSpeed { get; set; }
    	public string? WindDirection { get; set; }

    	public DateTime? MeasurementTime { get; set; }
    	public double? Humidity { get; set; }
    	public double? AtmosphericPressure { get; set; }
    	public double? Precipitation { get; set; }          // mm 
    	public double? PrecipitationPerHour { get; set; }   // mm/h
    	public double? AirTemperature { get; set; }

}

File: eco-map/src/types/Measurement.ts


export interface Measurement {
    id: number;
    clientId?: string;
    locationId: number;
    // Стабильная ссылка на clientId родительской локации. Нужна офлайн-очереди,
    // чтобы связать замер с локацией, которая ещё не синхронизирована с сервером.
    locationClientId?: string;

    sensorName: string;
    comment?: string | null;

    o2?: number | null;
    co?: number | null;
    sO2?: number | null;
    no?: number | null;
    ch?: number | null;
    cO2?: number | null;
    nO2?: number | null;
    h2CO?: number | null;

    pM25?: number | null;
    pM10?: number | null;
    tvoc?: number | null;

    windSpeed?: number | null;
    windDirection?: string | null;

    creationDate: string;
    updatedAt?: string | null;
    deletedAt?: string | null;
    measurementTime?: string | null;

    humidity?: number | null;
    atmosphericPressure?: number | null;

    precipitation?: number | null;
    precipitationPerHour?: number | null;

    airTemperature?: number | null;
}
